mypy/config_parser.py

```diff
@@ -60,14 +60,19 @@ def parse_version(v: str | float) -> tuple[int, int]:
     return major, minor
 
 
-def try_split(v: str | Sequence[str], split_regex: str = "[,]") -> list[str]:
-    """Split and trim a str or list of str into a list of str"""
+def try_split(v: str | Sequence[str] | Any, split_regex: str = "[,]") -> list[str] | Any:
+    """Split and trim a str or sequence (eg: list) of str into a list of str.
+    Non-str elements will simply be returned untouched. Feel free to one day
+    fix the typing of the calling code, and remove this caveat and Any."""
     if isinstance(v, str):
         items = [p.strip() for p in re.split(split_regex, v)]
         if items and items[-1] == "":
             items.pop(-1)
         return items
-    return [p.strip() for p in v]
+    elif isinstance(v, Sequence):
+        return [p.strip() if isinstance(p, str) else p for p in v]
+    else:
+        return v
```

Comment:
To keep the blast radius minimal, maybe 

```suggestion
        raise argparse.ArgumentTypeError(f"Expected a list or a stringified version thereof, but got: '{v}'")
```

Reply:
Then you won't need `Any` in return type too